presto-native-execution/presto_cpp/main/operators/BroadcastFactory.cpp

```diff
@@ -149,17 +168,23 @@ velox::BufferPtr BroadcastFileReader::next() {
   if (!hasNext()) {
     return nullptr;
   }
-
-  auto readFile = fileSystem_->openFileForRead(broadcastFileInfo_->filePath_);
-  auto buffer = AlignedBuffer::allocate<char>(readFile->size(), pool_, 0);
-  readFile->pread(0, readFile->size(), buffer->asMutable<char>());
-  numBytes_ += readFile->size();
+  auto inputStream = bufferedInput_->read(
+      0, numBytes_, velox::dwio::common::MetricsLog::MetricsType::FILE);
+  auto buffer = AlignedBuffer::allocate<char>(numBytes_, pool_, 0);
+  inputStream->readFully(buffer->asMutable<char>(), numBytes_);
   hasData_ = false;
   return buffer;
 }
 
 folly::F14FastMap<std::string, int64_t> BroadcastFileReader::stats() {
-  return {{"broadcastExchangeSource.numBytes", numBytes_}};
+  auto cacheStats = asyncDataCache_->refreshStats();
+  return {
+      {"broadcastExchangeSource.numBytes", numBytes_},
+      {"broadcastExchangeSource.cacheHit", cacheStats.numHit},
```

Comment:
The cache is shared so we can't use the stats from async data cache to set operator's cache stats.

Reply:
removed.